### PR TITLE
Rebalance some traits

### PR DESCRIPTION
**Pull request by Coolthulhu**
Conservatively rebalanced some traits, mostly the starting ones. Mostly 1 point adjustments, even when the trait in question would deserve 2 point changes.

When balancing, it is often a good idea to compare point values to dexterity. If putting the points in dexterity would certainly be better than taking the trait, the trait is not worth its value in points.

Human traits:
- Light eater is not a very good trait, except in specialized scenarios like 0.01 item spawn winter. 3->2
- Accomplished sleeper was nearly useless. I boosted its effect 3 times. It still isn't worth it, but I didn't want to give it 50 "sleepy points". Maybe needs something more than pure "sleepy points" value? Also, 2->1
- Tough: high hp doesn't mean much (pain does mean much more) and 3 strength is MUCH better than this trait. Still not a good trait after changing 3->2
- Thick-skinned is either 1 or 2 points of cut resistance. It's extremely bad and only makes sense for nudist challenges (even then it would be worth 0.5 point). 2->1
- Deft is really bad AND made obsolete by Brawling. Should probably be removed altogether. 2->1
- Nobody takes drunken master. It takes alcohol, so it requires midgame equimpent. 2->1
- Addiction resistant: it's easy not to get addicted and also easy to wait addictions out. 3->2
- Terrifying: nearly worthless. 2->1
- High adrenaline is the first trait to get a 2 point discount because it is just soooo bad. It only activates when you're too damaged to keep fighting AND gets obsolete in midgame. 3->1
- Inconspicuous doesn't really do much nowadays. 2->1
- Skilled liar affects those NPC interactions which can be cheesed easily. 2->1
- Masochist is simply too weak. 2->1
- Animal discord is a free point. Was 2 free points. Infighting made it free-er 2 points, but it was a no-brainer even before. -2->-1
- Slow runner was much a much stronger penalty than all other 2 point ones. I'd say 4 points would fit it more, but went conservatively with 3. -2->-3
- Trigger happy affects only a small selection of firearms and it's easy to just not use those. -2->-1
- Illiterate is a game changer. -5->-6, but should be -12 (not exaggerating here). Trait points are cheap, lack of literacy bans survivors from the best recipes.

Mutant starting traits (Experiment or Lab scenarios):
- Toe talon is amazing. -2->-1, but should probably be -2->+1
- Beak forbids gas mask, making all gas zombies a problem. -1->-3
- Gills only work in water and there isn't much to find there. 3->2
- Webbed hand was abysmal at -1 - it penalizes dex and forbids gloves just for a swimming speed bonus. -1->-2 and changed wetness resistance to wetness friendliness
- Genetically unstable is fun to play with and with Robust Genetics is actually really, really good. Even without it, a lot of mutations are a great boon to the survivor. -4->-2
- Minor radioactivity doesn't do much. Even now that it actually does SOMETHING. -4->-2
- Carnivore is worse than fruit hate, veggy hate and wheat allergy combined. -3->-4 (but should be much more and just cancel all the plant hate traits)

Also rebalanced some mutations where point values are irrelevant. Maybe someday someone will want to make a scenario that includes heavy mutations. Though then we'd need to rebalance all the other traits with zero balance.

### CODE REVIEW

**Comment by Rivet-the-Zombie** [comment]
I like this.

**Comment by Lain-** [comment]
Out of the scope of the PR here, but I think traits/skill/stat selection could use a similar approach in PR #11497 . Multiply everything by 10 so we can have more gradual balacing.

**Comment by Zireael07** [comment]
@Lain: Agreed.

@Coolthulhu : Good thinking on carnivore cancelling plant hate traits.

**Comment by PropaneSoup** [comment]
I'm not 100% sold on multiplying everything by 10 for traits, but I do agree that this PR is a good step towards making more traits useful.

**Comment by Snaaty** [comment]
The balancing seems good to me. Multiplying everything by ten is probably hard to do, as the same points are being used to get stats and professions.

**Comment by Coolthulhu** [comment]
It wouldn't be hard (simple value replacement), but it could lead to traits being rebalanced by changing their point values rather than their effects. Traits like Thick Skin should get buffed to be worth 1 point rather than balanced by lowering its point cost to a fraction.

I didn't balance mechanics behind many of those traits (even those that should get it), because I didn't want to make giant changes in a single PR. I intend to follow up with more balancing PRs, especially for the really weak stuff.
Eventually traits like Deft should get an effect (better crit chance?) that would make them worth their cost.

**Comment by graysage1** [comment]
@Coolthulhu I strongly agree with your comments.  However, in this case:

>High adrenaline is the first trait to get a 2 point discount because it is just soooo bad. It only activates when you're too damaged to keep fighting AND gets obsolete in midgame. 3->1

Is it possible to make "high adrenaline" more interesting instead?  It's more work, but wouldn't it be better to have a fun trait instead of a nearly useless one that will probably never kick in, even if it is cheap?  How about making high adrenaline kick in if any body part drops below 60% hp or takes more than 15% hp damage in a single hit?

**Comment by Coolthulhu** [comment]
@graysage1 Yes, it is possible to rebalance it. I didn't play with the trait long enough to get a good idea on how would I do that, though. It isn't a simple numerical trait you can just increase or lower.

One of the reasons to lower the point costs of bad traits is to make players take them and get familiar with them - that way someone can offer valid criticism and suggest how it should work.

In the end it should probably: counter pain (preferably without using painkillers), activate before it's too late, last long enough to escape/kill stuff, maybe activate without getting damaged (just a lot of tough stuff nearby) and maybe even save the player from death unless HP reaches high negatives.

I'd rather delay fiddling with adrenaline for now. Kevin wants to introduce fatigue mechanics (real fatigue, not sleepiness), so adrenaline mechanics will have to be changed anyway.

**Comment by KA101** [comment]
I don't appreciate this being done on a straight "DX or this" basis, as that encourages point-optimization.  Would have liked to have heard about it earlier as well.

**Comment by Coolthulhu** [comment]
Point optimization is the ultimate goal of balancing and there is no going around it.
I went with small, conservative changes: if I couldn't think of a regular, not-challenge situation where the trait would be useful for those points, I lowered the cost. And the reverse: if the only reason not to pick the trait was that there are better "free point" traits, I nerfed it too.

Dex vs trait is for the minimum point value. If there is no reason to ever take the trait over dex (Deft), the trait is simply bad and there's no masquerading it as a design choice. Some traits are more or less bad - Light Eater for example can be useful for new players who don't know how to cook, but for everyone else (except for challenges) it simply wasn't worth the 3 points.

**Comment by KA101** [comment]
Or it could be that DX is overpowered.  I'm not terribly interested in perfect balancing.

**Comment by Coolthulhu** [comment]
Perfect balancing isn't possible, but many of the traits were simply too weak to be taken. And others too good not to be taken.
What's wrong with the idea of perfect point balance anyway? If a trait is worth 3 points, why shouldn't it cost 3 points?
If it went the other way - a point costing 2 points was balanced to be worth precisely 2 points - I could understand. But traits shouldn't be so weak that they are only taken as a challenge (unless they're intended to, like illiterate) or so strong they are only not-taken as a challenge (no trait should be a no-brainer and especially no trait should be intended to be a no-brainer).

**Comment by graysage1** [comment]
@Coolthulhu 

>One of the reasons to lower the point costs of bad traits is to make players take them and get familiar with them - that way someone can offer valid criticism and suggest how it should work.

This makes a lot of sense to me.

**Comment by Marrim** [comment]
Are you sure about Light Eater? It is incredibly useful because you save on carrying capacity. Plus the convenience of having to stop to eat less often.

Also, Less Sleep is totally a no-brainer.

**Comment by Coolthulhu** [comment]
Could nerf back light eater, but I'm pretty sure the convenience of it is far outweighed by its crippling cost. It is clearly worse than Quick, Fleet-footed or Fast learner, for example.
Plus, tanking hunger is at the moment very easy - you can easily go famished with no significant ill effects.
It can be useful, but not 3 points useful.

Less sleep is not a no-brainer, because it makes it significantly harder to heal early on and later on becomes obsolete due to (atomic) coffee.

**Comment by KA101** [comment]
OK, so we increase the problems for hunger.

I'm thinking some of these traits need reworked--Inconspicuous, for example, is obsolete w/o dynamic spawn; I'm thinking that it should give critters a penalty to see you--and I'd prefer that to messing around with the point values, especially when I've not had the time to go through and give this a hard review.

**Comment by Coolthulhu** [comment]
I agree about Inconspicuous - it needs a rework, plus your idea is quite good. However as it stands, there are many skills that could use just a simple point buff/nerf.

Many of those skills simply need testing. High point costs heavily discourage taking skills with no obvious benefits. Someone who takes High adrenaline once will probably not take it a second time after not noticing any effects. If it did cost 1 point, it wouldn't scare away experimentation.

Deft, Inconspicuous and Terrifying should get reworked completely, but traits like Tough or Masochist could just get a point buff. Their effects aren't mechanically unsound, just heavily overpriced.

File: src/player.cpp [hunk 1]
```diff
@@ -11414,7 +11417,12 @@ bool player::can_sleep()
         sleepy += int((fatigue - 192) / 16);
     }
     sleepy += rng(-8, 8);
-    sleepy -= 2 * stim;
+    if( !has_trait("INSOMNIA") ) {
+        sleepy -= 2 * stim;
+    } else {
+        // Make it harder for insomniac to get around the trait
+        sleepy -= stim;
+    }
```
Comment (Marrim) [inline src/player.cpp]: Doesn't that also incidentally make positive stim less of an issue when trying to sleep as insomniac?
Perhaps
if( !has_trait("INSOMNIA") || ( stim > 0 ) ) {

would work better?

**Pull request merged**